sonic_y_cable/y_cable.py

```diff
@@ -0,0 +1,347 @@
+#!/usr/bin/env python
+#
+# y_cable.py
+#
+#  definitions for implementing Y cable access and configurations
+#   API's for Y cable functionality in SONiC
+
+try:
+    import binascii
+    import sys
+
+    import sonic_platform.platform
+    from sonic_py_common import logger
+
+except ImportError as e:
+    raise ImportError(str(e) + " - required module not found")
+
+# Global logger instance for helper functions and classes to log
+helper_logger = logger.Logger(SYSLOG_IDENTIFIER)
+
+# Global platform_chassis instance to call get_sfp required for read/write eeprom
+platform_chassis = None
+
+# definitions of the offset with width accomodated for values
+# of MUX register specs of upper page 0x04 starting at 640
+# info eeprom for Y Cable
+Y_CABLE_IDENTFIER_LOWER_PAGE = 0
+Y_CABLE_IDENTFIER_UPPER_PAGE = 128
+Y_CABLE_DETERMINE_CABLE_READ_SIDE = 640
+Y_CABLE_CHECK_LINK_ACTIVE = 641
+Y_CABLE_SWITCH_MUX_DIRECTION = 642
+Y_CABLE_ACTIVE_TOR_INDICATOR = 645
+Y_CABLE_MANUAL_SWITCH_COUNT = 669
+
+
+def load_platform_chassis_for_y_cable():
+    global platform_chassis
+
+    """
+    This call is neccessary to load and store the global platform_chassis
+    variable which provides the utiliy to call the Sfp_Base's class
+    apis of reading and writing eeprom which are required for API's
+    of the Y cable.
+    """
+
+    try:
+        platform_chassis = sonic_platform.platform.Platform().get_chassis()
+        helper_logger.log_info("chassis loaded {}".format(platform_chassis))
+    except Exception as e:
+        helper_logger.log_warning("Failed to load chassis due to {}".format(repr(e)))
```

Comment:
Would it be possible to prevent the need to expose this function to the caller? E.g., since `platform_chassis` is needed by all the functions in the file, maybe we should just move this try/except block out of a function so that it's done once immediately when the file is imported? Thoughts?

Reply:
Yes, that is what I was initially thinking of doing, the only thought was if platform_chassis should be available when cli/xcvrd will import this file. 
I will move it out side the function then.

Reply:
fixed